File: main-workshop/services/typescript/payment-processor/src/PaymentProcessor.ts
import { Logger } from '@aws-lambda-powertools/logger';
import type { DriverAssignedEvent, PaymentResult } from './models';
import { PaymentService } from './services/PaymentService';

/**
 * This file is part of the Payment Processor service.
 * Update this file in the Module 2 Idempotency exercise.
 */

let logger: Logger;

export class PaymentProcessor {
  private readonly paymentService: PaymentService;

  constructor(loggerInstance?: Logger) {
    logger = loggerInstance || new Logger();
    this.paymentService = new PaymentService();
  }

  async handlePayment(
    driverEvent: DriverAssignedEvent
  ): Promise<PaymentResult> {
    const result = await this.paymentService.processPayment(driverEvent);

    logger.info('Payment processing result', { result });

    if (result.success && result.payment) {
      logger.info('Payment created', {
        ride_id: result.payment?.rideId,
        payment_id: result.payment?.paymentId,
        payment_amount: result.payment?.amount,
        payment_method: result.payment?.paymentMethod,
      });
    }
    return result;
  }
}

File: main workshop/services/java/payment-processor/src/main/java/com/powertoolsride/paymentprocessor/PaymentProcessor.java
package com.powertoolsride.paymentprocessor;

import com.powertoolsride.paymentprocessor.model.DriverAssignedEvent;
import com.powertoolsride.paymentprocessor.model.PaymentResult;
import com.powertoolsride.paymentprocessor.service.PaymentService;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.slf4j.MDC;

import java.util.HashMap;
import java.util.Map;

import static software.amazon.lambda.powertools.logging.argument.StructuredArguments.entries;

public class PaymentProcessor {
    private static final Logger LOGGER = LoggerFactory.getLogger(PaymentProcessor.class);
    private final PaymentService paymentService;

    public PaymentProcessor() {
        this.paymentService = new PaymentService();
    }

    public PaymentResult processPayment(DriverAssignedEvent driverEvent) {
        PaymentResult result = paymentService.processPayment(driverEvent);

        if (result.success()) {
            MDC.put("ride_id", result.payment().rideId());
            MDC.put("rider_id", result.payment().riderId());
            MDC.put("payment_id", result.payment().paymentId());
            MDC.put("payment_amount", result.payment().amount().toString());
            MDC.put("payment_method", result.payment().paymentMethod());

            LOGGER.info("Payment created");
        }

        return result;
    }
}
